compparallel.go

```diff
@@ -0,0 +1,267 @@
+package routinghelpers
+
+import (
+	"context"
+	"errors"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/ipfs/go-cid"
+	"github.com/libp2p/go-libp2p-core/peer"
+	"github.com/libp2p/go-libp2p-core/routing"
+)
+
+var _ routing.Routing = &Parallel{}
+
+type ComposableParallel struct {
+	routers []*ParallelRouter
+}
+
+func NewComposableParallel(routers []*ParallelRouter) *ComposableParallel {
+	return &ComposableParallel{
+		routers: routers,
+	}
+}
+
+func (r *ComposableParallel) Provide(ctx context.Context, cid cid.Cid, provide bool) error {
+	var wg sync.WaitGroup
+	errCh := make(chan error)
+	closeCh := make(chan bool)
+	for _, r := range r.routers {
+		r := r
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			select {
+			case <-ctx.Done():
+				if ctx.Err() != nil && !r.IgnoreError {
+					errCh <- ctx.Err()
+				}
+			case <-time.After(r.ExecuteAfter):
+				ctx, cancel := context.WithTimeout(ctx, r.Timeout)
+				defer cancel()
+				err := r.Router.Provide(ctx, cid, provide)
+				if err != nil &&
+					!errors.Is(err, routing.ErrNotSupported) &&
```

Comment:
This is interesting and I haven't fully thought through the implications. Why would we want to swallow this instead of yelling at the user that they've misconfigured their routers?

Reply:
I thought that failing the whole operation because some router method is not supported will be a PITA for the user, but we'll have a versatile enough configuration to avoid that. I'll remove the check.